feat(applications-exchange-cma): added doc page and enhance other exchange doc pages CTOR-1928 (#4808)
centreon/centreon-documentation: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/applications-exchange-cma.md

---
id: applications-exchange-cma
title: Microsoft Exchange CMA
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> 

Reviewer: EN in FR

## Connector dependencies

The following monitoring connectors will be installed when you install the **Microsoft Exchange CMA** connector through the

> **Configuration > Monitoring Connector Manager** menu:

Reviewer: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu:
```

| Exchange-Services | App-Exchange-Services-CMA-custom  | Check exchange services  |
| Queues            | App-Exchange-Queues-CMA-custom    | Check queue status       |

> The services listed above are created automatically when the **App-Exchange-CMA-custom** host template is used.

</TabItem>
<TabItem value="Not attached to a host template" label="Not attached to a host template">

| Service Alias       | Service Template                            | Service Description                                    |
|:--------------------|:--------------------------------------------|:-------------------------------------------------------|
| Activesync-Mailbox  | App-Exchange-Activesync-Mailbox-CMA-custom  | Check activesync to a mailbox                          |
| Imap-Mailbox        | App-Exchange-Imap-Mailbox-CMA-custom        | Check IMAP to a mailbox                                |
| Mailboxes           | App-Exchange-Mailboxes-CMA-custom           | Check mailboxes (quota and user mailboxes by database) |
| Mapi-Mailbox        | App-Exchange-Mapi-Mailbox-CMA-custom        | Check MAPI to a mailbox                                |

> | Outlook-Webservices | App-Exchange-Outlook-Webservices-CMA-custom | Check outlook autodiscovery webservices                |

Reviewer: ```suggestion
| Outlook-Webservices | App-Exchange-Outlook-Webservices-CMA-custom | Check Outlook autodiscovery webservices                |
```

| Owa-Mailbox         | App-Exchange-Owa-Mailbox-CMA-custom         | Check OWA connection to a mailbox                      |
| Replication-Health  | App-Exchange-Replication-Health-CMA-custom  | Check replication health                               |

> The services listed above are not created automatically when a host template is applied. To use them, [create a service manually](/docs/monitoring/basic-objects/services), then apply the service template you want.

</TabItem>
</Tabs>

### Collected metrics & status

Here is the list of services for this connector, detailing all metrics and statuses linked to each service.

<Tabs groupId="sync">
<TabItem value="Activesync-Mailbox" label="Activesync-Mailbox">

> No metric for this service.

Reviewer: ```suggestion
No metrics for this service.
```

</TabItem>
<TabItem value="Databases" label="Databases">

| Name                                          | Unit  |
|:----------------------------------------------|:------|
| databases.space.size.bytes                    | B     |
| databases.space.available.bytes               | B     |
| status                                        | N/A   |
| *databases*~database.space.size.bytes         | B     |
| *databases*~database.space.available.bytes    | B     |
| mapi                                          | N/A   |
| mailflow                                      | N/A   |
| *databases*~database.mailflow.latency.seconds | s     |
| copystatus                                    | N/A   |

</TabItem>
<TabItem value="Exchange-Services" label="Exchange-Services">

> No metric for this service.

Reviewer: ```suggestion
No metrics for this service.
```

</TabItem>
<TabItem value="Imap-Mailbox" label="Imap-Mailbox">

> No metric for this service.

Reviewer: ```suggestion
No metrics for this service.
```

</TabItem>
<TabItem value="Mailboxes" label="Mailboxes">

| Name                                                     | Unit  |
|:---------------------------------------------------------|:------|
| *databases*#database.mailboxes.total.count               | count |
| *users*#user.soft.limit.count                            | count |
| *users*#user.hard.limit.count                            | count |
| *users*#user.quota.unlimited.limit.count                 | count |
| *users*#user.total.count                                 | count |
| *publicfolders*#publicfolder.soft.limit.count            | count |
| *publicfolders*#publicfolder.hard.limit.count            | count |
| *publicfolders*#publicfolder.quota.unlimited.limit.count | count |
| *publicfolders*#publicfolder.total.count                 | count |

</TabItem>
<TabItem value="Mapi-Mailbox" label="Mapi-Mailbox">

> No metric for this service.

Reviewer: ```suggestion
No metrics for this service.
```

</TabItem>
<TabItem value="Outlook-Webservices" label="Outlook-Webservices">

> No metric for this service.

Reviewer: ```suggestion
No metrics for this service.
```

</TabItem>
<TabItem value="Owa-Mailbox" label="Owa-Mailbox">

> No metric for this service.

Reviewer: ```suggestion
No metrics for this service.
```

</TabItem>
<TabItem value="Queues" label="Queues">

| Name   | Unit  |
|:-------|:------|
| status | N/A   |

</TabItem>
<TabItem value="Replication-Health" label="Replication-Health">

> No metric for this service.

Reviewer: ```suggestion
No metrics for this service.
```

</TabItem>
</Tabs>

## Prerequisites

### Network flow

Only one TCP flow must be open from the host to the poller.

| Source         | Destination | Protocol | Port | Purpose                                              |
|----------------|-------------|----------|------|------------------------------------------------------|

> | Monitored host | Collecteur  | TCP      | 4317 | Configuration retrieval, and OpenTelemetry data flow |

Reviewer: ```suggestion
| Monitored host | Poller | TCP      | 4317 | Configuration retrieval, and OpenTelemetry data flow |
```

### System prerequisites on the poller

> To be able to use the Centreon Monitoring agent, you must use a poller with at least version `24.09.0` for Centreon Cloud users and version `24.04.6` or `24.10.0` for On Prem users of `centreon-engine`. The Centreon Monitoring agent will configure itself by connecting to Centreon Engine.

### Configure Engine

[Configure how the poller and the agent will communicate](../getting-started/how-to-guides/cma/cma-setup.md#configure-polleragent-communication).

### System prerequisites on the monitored host

> The installer can be downloaded from the [centreon-collect's releases page](https://github.com/centreon/centreon-collect/releases?q=centreon-collect&expanded=true).

Reviewer: ```suggestion
The installer can be downloaded from the [centreon-collect's release page](https://github.com/centreon/centreon-collect/releases?q=centreon-collect&expanded=true).
```

#### Installing the Centreon Monitoring Agent

The installation and configuration procedure of Centreon Monitoring Agent for Windows is detailed in 
[this dedicated page](../getting-started/how-to-guides/cma/cma-setup.md#step-3-prepare-the-host).

## Installing the monitoring connector

### Pack

1. If the platform uses an *online* license, you can skip the package installation
instruction below as it is not required to have the connector displayed within the

> **Configuration > Monitoring Connector Manager** menu.

Reviewer: ```suggestion
**Configuration > Connectors > Monitoring Connectors** menu.
```

If the platform uses an *offline* license, install the package on the **central server**
with the command corresponding to the operating system's package manager:

<Tabs groupId="sync">
<TabItem value="Alma / RHEL / Oracle Linux 8" label="Alma / RHEL / Oracle Linux 8">

```bash
dnf install centreon-pack-applications-exchange-cma
```

</TabItem>
<TabItem value="Alma / RHEL / Oracle Linux 9" label="Alma / RHEL / Oracle Linux 9">

```bash
dnf install centreon-pack-applications-exchange-cma
```

</TabItem>
<TabItem value="Debian 11 & 12" label="Debian 11 & 12">

```bash
apt install centreon-pack-applications-exchange-cma
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-pack-applications-exchange-cma
```

</TabItem>
</Tabs>

2. Whatever the license type (*online* or *offline*), install the **Microsoft Exchange CMA** connector through

> the **Configuration > Monitoring Connector Manager** menu.

Reviewer: ```suggestion
the **Configuration > Connectors > Monitoring Connectors** menu.
```

| CRITICAL     | Set critical threshold. You can use the following variables: %\{result\}, %\{scenario\}                                                | not %\{result\} =~ /Success/i |           |
| WARNING      | Set warning threshold. You can use the following variables: %\{result\}, %\{scenario\}                                                 |                               |           |
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). | --verbose                     |           |

</TabItem>
<TabItem value="Mailboxes" label="Mailboxes">

| Macro                               | Description                                                                                                                            | Default value | Mandatory |
|:------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------|:--------------|:---------:|

> | PSERVER                             | Select mailboxes by an uniq server name (In Powershell)                                                                                |               |           |

Reviewer: ```suggestion
| PSERVER                             | Select mailboxes by a unique server name (in Powershell)                                                                                |               |           |
```

> | PSDATABASE                          | Select mailboxes by an uniq database name (In Powershell)                                                                              |               |           |

Reviewer: ```suggestion
| PSDATABASE                          | Select mailboxes by a unique database name (in Powershell)                                                                              |               |           |
```

| CRITICAL     | Set critical threshold. You can use the following variables: %\{type\}, %\{id\}, %\{message\}                                          | not %\{type\} =~ /Success\|Information/i |           |
| WARNING      | Set warning threshold. You can use the following variables: %\{type\}, %\{id\}, %\{message\}                                           |                                          |           |
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). | --verbose                                |           |

</TabItem>
<TabItem value="Owa-Mailbox" label="Owa-Mailbox">

| Macro        | Description                                                                                                                            | Default value                 | Mandatory |
|:-------------|:---------------------------------------------------------------------------------------------------------------------------------------|:------------------------------|:---------:|

> | URL          | Set the OWA Url                                                                                                                        |                               |     X     |

Reviewer: ```suggestion
| URL          | Set the OWA Url                                                                                                                        |                               |     X     |
```
```suggestion
| URL          | Set the OWA URL                                                                                                                       |                               |     X     |
```

| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). | --verbose                    |           |

</TabItem>
</Tabs>

3. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The service appears in the list of services, and on the **Resources Status** page. The command that is sent by the connector is displayed in the details panel of the service: it shows the values of the macros.

## How to check in the CLI that the configuration is OK and what are the main options for?

Once the plugin is installed, log into your Centreon poller's CLI using the
**centreon-engine** user account (`su - centreon-engine`). Test that the connector 
is able to monitor a resource using a command like this one (replace the sample values by yours):

```bash
"/centreon_plugins.exe" \
    --plugin=apps::microsoft::exchange::local::plugin \
    --mode=queues \
	--timeout=120 \
	--warning-status="" \
	--critical-status="not %\{status\} =~ /Ready|Active/i" \
	--verbose
```

> 

Reviewer: isn't it wrong?

The expected command output is shown below:

```bash
OK: All queues are ok 
```

### Troubleshooting

Please find the [troubleshooting documentation](../getting-started/how-to-guides/troubleshooting-plugins.md)
for Centreon Plugins typical issues.

### Available modes

In most cases, a mode corresponds to a service template. The mode appears in the execution command for the connector.
In the Centreon interface, you don't need to specify a mode explicitly: its use is implied when you apply a service template.
However, you will need to specify the correct mode for the template if you want to test the execution command for the 
connector in your terminal.

All available modes can be displayed by adding the `--list-mode` parameter to
the command:

```bash

> "/centreon_plugins.exe" \

Reviewer: is this correct?

Author: ```suggestion
"C:\Program Files\Centreon\Plugins\centreon_plugins.exe" \
```

| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). Toutes les options sont listées [ici](#options-disponibles). | --verbose                    |             |

</TabItem>
</Tabs>

3. [Déployez la configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). Le service apparaît dans la liste des services supervisés, et dans la page **Statut des ressources**. La commande envoyée par le connecteur est indiquée dans le panneau de détails du service : celle-ci montre les valeurs des macros.

## Comment puis-je tester le plugin et que signifient les options des commandes ?

Une fois le plugin installé, vous pouvez tester celui-ci directement en ligne
de commande depuis votre collecteur Centreon en vous connectant avec
l'utilisateur **centreon-engine** (`su - centreon-engine`). Vous pouvez tester
que le connecteur arrive bien à superviser une ressource en utilisant une commande
telle que celle-ci (remplacez les valeurs d'exemple par les vôtres) :

```bash

> "/centreon_plugins.exe" \

Author: ```suggestion
"C:\Program Files\Centreon\Plugins\centreon_plugins.exe" \
```
